ArjunaJTA/jta/classes/com/arjuna/ats/jta/logging/RecoveryRequired.java

```diff
@@ -0,0 +1,14 @@
+package com.arjuna.ats.jta.logging;
+
+public class RecoveryRequired {
+
+    private static boolean recoveryProblems;
```

Comment:
As this is a flag which is used to mark if a recovery problem occurs and this flag could be accessed from different threads and value could be set from the different threads then the code has to(!) consider that. The property should be of type atomic.

Reply:
Sorry I missed this thought. Thanks for pointing this out. I will make it thread safe.